### PR TITLE
[RFC] win: test: enable more non-legacy functional tests on Windows

### PR DESCRIPTION
**Pull request by janlazo**
Cherry-picked from #7412 
The patch enables some functional tests with minimal changes to work in Windows.

### CODE REVIEW

**Comment by janlazo** [comment]
QuickBuild failure seems unrelated to this PR since I didn't work on the tui.

**Comment by justinmk** [comment]
Can you squash the "yes is unavailable" commits into 1 ?

File: test/functional/eval/system_spec.lua [hunk 1]
```diff
@@ -89,7 +89,9 @@ describe('system()', function()
     end)
 
     it('does NOT run in shell', function()
-      if not iswin() then
+      if iswin() then
+        eq("C:\\Windows\\System32\\cmd.exe\n", eval("system(['where', 'cmd'])"))
```
Comment (justinmk) [inline test/functional/eval/system_spec.lua]: This isn't testing anything useful

File: test/functional/eval/system_spec.lua [hunk 2]
```diff
@@ -89,7 +89,9 @@ describe('system()', function()
...
+      else
         eq("* $PATH %PATH%\n", eval("system(['echo', '*', '$PATH', '%PATH%'])"))
```
Comment (justinmk) [inline test/functional/eval/system_spec.lua]: Why v doesn't this work on Windows?

File: test/functional/terminal/ex_terminal_spec.lua [hunk 3]
```diff
@@ -38,7 +41,11 @@ describe(':terminal', function()
   end)
 
   it("in normal-mode :split does not move cursor", function()
-    feed_command([[terminal while true; do echo foo; sleep .1; done]])
+    if iswin() then
+      feed_command([[terminal for /L \\%I in (1,0,2) do echo \\%I]])
+    else
```
Comment (justinmk) [inline test/functional/terminal/ex_terminal_spec.lua]: Was this a copy-paste mistake?

Comment (justinmk) [reply test/functional/terminal/ex_terminal_spec.lua]: I mean the line below this

Comment (janlazo) [reply test/functional/terminal/ex_terminal_spec.lua]: Yes

File: test/functional/eval/system_spec.lua (hunk 2 above)
Comment (janlazo) [reply test/functional/eval/system_spec.lua]: echo.exe (mingw) is not in PATH so `v:shell_error` will be set.

File: test/functional/eval/system_spec.lua (hunk 1 above)
Comment (janlazo) [reply test/functional/eval/system_spec.lua]: I can remove this and disable the test on Windows if that's better.
I assumed that this test should not use `&shell`

Comment (justinmk) [reply test/functional/eval/system_spec.lua]: Yes but the test is supposed to check that env vars aren't expanded

**Comment by justinmk** [comment]
Is this note still relevant?

> In Windows, :w! is required to overwrite files for test on FileWriteCmd. Removing the files via os.remove() or 'del' command in cmd.exe has no effect.

(No need to push an update, I'm merging it locally now)

**Comment by justinmk** [comment]
merged. thanks a lot!

**Comment by janlazo** [comment]
Yes, because it happens on Vim too. I don't know if that's standard behaviour on Windows so I disabled the test for now.

**Comment by justinmk** [comment]
Oh, I removed the note from the commit message. Maybe you could add it again in one of the other PRs.

**Pull request closed without merging**